pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,511 @@
+	l := p.log.With(keyIDTag, *awsKeyID, aliasTag, alias)
+	describeResp, err := p.kmsClient.DescribeKeyWithContext(ctx, &kms.DescribeKeyInput{KeyId: alias})
+	if err != nil {
+		return nil, kmsErr.New("failed to describe key: %v", err)
+	}
+
+	if !*describeResp.KeyMetadata.Enabled {
+		l.Debug("Skipped disabled key")
+		return nil, nil
+	}
+
+	spireKeyID, err := p.spireKeyIDFromAlias(*alias)
+	if err != nil {
+		l.Debug("Skipped key", "reason", err)
+		return nil, nil
+	}
+
+	keyType, err := keyTypeFromKeySpec(*describeResp.KeyMetadata.CustomerMasterKeySpec)
+	if err != nil {
+		l.Debug("Skipped key", "reason", err)
+		return nil, nil
+	}
+
+	getPublicKeyResp, err := p.kmsClient.GetPublicKeyWithContext(ctx, &kms.GetPublicKeyInput{KeyId: alias})
+	if err != nil {
+		return nil, kmsErr.New("failed to get public key: %v", err)
+	}
+
+	return &keyEntry{
+		KMSKeyID: *awsKeyID,
+		Alias:    *alias,
+		PublicKey: &keymanager.PublicKey{
+			Id:       spireKeyID,
+			Type:     keyType,
+			PkixData: getPublicKeyResp.PublicKey,
+		},
+	}, err
+}
+
+func (p *Plugin) fetchAliasesPage(ctx context.Context, marker *string) (*string, error) {
+	aliasesResp, err := p.kmsClient.ListAliasesWithContext(ctx, &kms.ListAliasesInput{
+		Marker: marker,
+	})
+	if err != nil {
+		return nil, kmsErr.New("failed to fetch keys: %v", err)
+	}
+
+	p.log.Debug(fmt.Sprintf("%v keys were found", len(aliasesResp.Aliases)))
+
+	for _, alias := range aliasesResp.Aliases {
+		if alias.AliasName == nil || alias.TargetKeyId == nil {
+			continue
+		}
+		l := p.log.With(keyIDTag, *alias.TargetKeyId, aliasTag, *alias.AliasName)
+		l.Debug("Processing key")
+		entry, err := p.buildKeyEntry(ctx, alias.AliasName, alias.TargetKeyId)
+		switch {
+		case err != nil:
+			return nil, kmsErr.New("failed to process KMS key: %v", err)
+		case entry != nil:
+			err := p.setEntry(entry.PublicKey.Id, *entry)
+			l.Debug("Added key")
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return aliasesResp.NextMarker, nil
+}
+
+func (p *Plugin) spireKeyIDFromAlias(alias string) (string, error) {
+	tokens := strings.SplitAfter(alias, p.keyPrefix)
+	if len(tokens) != 2 {
+		return "", fmt.Errorf("alias does not contain SPIRE prefix")
+	}
+
+	return tokens[1], nil
+}
+
+func (p *Plugin) aliasFromSpireKeyID(spireKeyID string) string {
+	return fmt.Sprintf("%v%v%v", aliasPrefix, p.keyPrefix, spireKeyID)
+}
+
+func (p *Plugin) descriptionFromSpireKeyID(spireKeyID string) string {
+	return fmt.Sprintf("%v%v", p.keyPrefix, spireKeyID)
+}
+
+// validateConfig returns an error if any configuration provided does not meet acceptable criteria
+func (p *Plugin) validateConfig(c string) (*Config, error) {
+	config := new(Config)
+
+	if err := hcl.Decode(config, c); err != nil {
+		return nil, kmsErr.New("unable to decode configuration: %v", err)
+	}
+
+	if config.Region == "" {
+		return nil, kmsErr.New("configuration is missing a region")
+	}
+
+	if config.AccessKeyID == "" {
```

Comment:
If not configured, we may try to get the access key ID and secret access key from `AWS_ACCESS_KEY_ID ` and `AWS_SECRET_ACCESS_KEY` env vars as we do in other plugins that have those values as a default value?

Reply:
@amartinezfayo just a heads up. When I updated the SDK to V2 I removed our custom implementation for reading the env vars. The reasoning is that the SDK will do it on its own https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#LoadDefaultConfig
If both env vars and the config file are set, the config file will take precedence.